Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -152,46 +152,44 @@ internal class ModuleMetadataEmitter(
-                            element.annotations.mapTo(km.annotations) { it.map() }
-                            km.returnType = element.returnType.map()
-                            km.uniqId = data.uniqIds.uniqIdForFunction(element)
-                        }
+                    val annotations = if (bridgeBuilderResult.nativeBridges.isSupported(element)) {
+                        element.annotations
                     } else {
-                        null
+                        listOf(AnnotationStub.Deprecated.unableToImport)
+                    }
+                    KmFunction(element.flags, element.name).also { km ->
+                        km.receiverParameterType = element.receiver?.type?.map()
+                        element.typeParameters.mapTo(km.typeParameters) { it.map() }
+                        element.parameters.mapTo(km.valueParameters) { it.map() }
+                        annotations.mapTo(km.annotations) { it.map() }
+                        km.returnType = element.returnType.map()
+                        km.uniqId = data.uniqIds.uniqIdForFunction(element)
                     }
                 }
 
         override fun visitProperty(element: PropertyStub, data: VisitingContext) =
                 data.withMappingExtensions {
-                    val kind = element.bridgeSupportedKind
-                    if (kind != null) {
-                        val name = getPropertyNameInScope(element, data.container)
-                        KmProperty(element.flags, name, kind.getterFlags, kind.setterFlags).also { km ->
-                            element.annotations.mapTo(km.annotations) { it.map() }
-                            km.uniqId = data.uniqIds.uniqIdForProperty(element)
-                            km.receiverParameterType = element.receiverType?.map()
-                            km.returnType = element.type.map()
-                            if (kind is PropertyStub.Kind.Var) {
-                                kind.setter.annotations.mapTo(km.setterAnnotations) { it.map() }
-                                // TODO: Maybe it's better to explicitly add setter parameter in stub.
-                                km.setterParameter = FunctionParameterStub("value", element.type).map()
-                            }
-                            km.getterAnnotations += when (kind) {
-                                is PropertyStub.Kind.Val -> kind.getter.annotations.map { it.map() }
-                                is PropertyStub.Kind.Var -> kind.getter.annotations.map { it.map() }
-                                is PropertyStub.Kind.Constant -> emptyList()
-                            }
-                            if (kind is PropertyStub.Kind.Constant) {
-                                km.compileTimeValue = kind.constant.mapToAnnotationArgument()
-                            }
+                    val (kind, annotations) = element.bridgeSupportedKind?.let { it to element.annotations }
```

Comment:
We might have an additional case with properties, when getter is available but setter is not. Could you check this?

Reply:
d761925f77cd8c22391aabd36074a3158b7c07fd